Explorer/Assets/DCL/PerformanceAndDiagnostics/Profiling/ECS/DebugViewProfilingSystem.cs

```diff
@@ -145,25 +158,70 @@ protected override void Update(float t)
                 }
 
-
-
                 if (frameTimingsEnabled && bottleneckDetector.IsFrameTimingSupported && bottleneckDetector.TryCapture())
                     UpdateFrameTimings();
 
                 framesSinceMetricsUpdate++;
             }
         }
 
+        private void SampleProfilerCounters()
+        {
+            totalHeapSize = 0ul;
+            totalHeapSizeExecutable = 0ul;
+            totalPhysicalSize = 0ul;
+            usedHeapSize = 0ul;
+            heapSizeLimit = 0ul;
+            totalExternalSize = 0ul;
+            activeEngines = 0;
+
+            SampleProfilerCountersQuery(World);
+
+            if (activeEngines > 0)
+                heapSizeLimit /= (ulong)activeEngines;
+
+#if ENABLE_PROFILER
+            JavaScriptProfilerCounters.TOTAL_HEAP_SIZE.Sample(totalHeapSize);
+            JavaScriptProfilerCounters.TOTAL_HEAP_SIZE_EXECUTABLE.Sample(totalHeapSizeExecutable);
+            JavaScriptProfilerCounters.TOTAL_PHYSICAL_SIZE.Sample(totalPhysicalSize);
+            JavaScriptProfilerCounters.USED_HEAP_SIZE.Sample(usedHeapSize);
+            JavaScriptProfilerCounters.TOTAL_EXTERNAL_SIZE.Sample(totalExternalSize);
+            JavaScriptProfilerCounters.ACTIVE_ENGINES.Sample(activeEngines);
+#endif
+        }
+
+        [Query]
+        private void SampleProfilerCounters(ISceneFacade scene0)
+        {
+            var scene = (SceneFacade)scene0;
+            var heapInfo = scene.runtimeInstance.RuntimeHeapInfo;
+
+            if (heapInfo != null)
+            {
+                totalHeapSize += heapInfo.TotalHeapSize;
+                totalHeapSizeExecutable += heapInfo.TotalHeapSizeExecutable;
+                totalPhysicalSize += heapInfo.TotalPhysicalSize;
+                usedHeapSize += heapInfo.UsedHeapSize;
+                heapSizeLimit += heapInfo.HeapSizeLimit;
+                totalExternalSize += heapInfo.TotalExternalSize;
+                activeEngines += 1;
+            }
+        }
+
         private void UpdateSceneMetrics()
         {
-            bool isCurrentScene = scenesCache is { CurrentScene: { SceneStateProvider: { IsCurrent: true } } };
-            JsMemorySizeInfo totalJsMemoryData = v8ActiveEngines.GetEnginesSumMemoryData();
-            JsMemorySizeInfo currentSceneJsMemoryData = isCurrentScene ? v8ActiveEngines.GetEnginesMemoryDataForScene(scenesCache.CurrentScene.Info) : new JsMemorySizeInfo();
-
-            jsHeapUsedSize.Value = $"{totalJsMemoryData.UsedHeapSizeMB:F1} | {currentSceneJsMemoryData.UsedHeapSizeMB:F1}";
-            jsHeapTotalSize.Value = $"{totalJsMemoryData.TotalHeapSizeMB:F1} | {currentSceneJsMemoryData.TotalHeapSizeMB:F1}";
-            jsHeapTotalExecutable.Value = $"{totalJsMemoryData.TotalHeapSizeExecutableMB:F1} | {currentSceneJsMemoryData.TotalHeapSizeExecutableMB:F1}";
-            jsHeapLimit.Value = $"{totalJsMemoryData.HeapSizeLimitMB:F1}";
+            V8RuntimeHeapInfo current = null;
```

Comment:
previous naming was more verbose, so let's keep it. Please rename it into `currentScene`